openmdao/utils/file_utils.py

```diff
@@ -190,6 +190,8 @@ def _load_and_exec(script_name, user_args):
     with open(script_name, 'rb') as fp:
         code = compile(fp.read(), script_name, 'exec')
 
+    os.environ['OPENMDAO_SCRIPT_NAME'] = script_name
```

Comment:
We probably want to wrap this in `with set_env_vars_context(OPENMDAO_SCRIPT_NAME=script_name)` in case there's some unforeseen side effect of the environment variable after this.